main.tf

```diff
@@ -56,8 +56,25 @@ variable "external_subnets" {
 }
 
 variable "availability_zones" {
-  description = "a comma-separated list of availability zones"
-  default     = "us-west-2a,us-west-2b,us-west-2c"
+  description = "a comma-separated list of availability zones, defaults to all AZ of the region"
+  default     = ""
+}
+
+variable "default_availability_zones" {
+  description = "a mapping of the default availability zones for each AWS region"
+
+  default = {
+    us-east-1      = "us-east-1a,us-east-1b,us-east-1c,us-east-1e"
```

Comment:
Nice this looks good.

This reminds me, we may want to re-work how we're doing the VPC networking in this case since we'd also need the CIDR blocks depending on the count of availability zones.

We could definitely achieve it, but we'll need a bit more to get there, something like:

```
count = "${length($split(",", var.availability_zones))}"
```
]https://github.com/segmentio/stack-old/pull/96/files